Source/Helpers/MessageSender.swift

```diff
@@ -15,22 +16,51 @@ public class MessageSender {
     private let queue = DispatchQueue(label: "MessageSender")
 
     private func send(message: Message, additionalData: Data?, to channel: Storage.Channel, date: Date) throws {
+        let card = try channel.getCard()
+
+        let ratchetChannel: RatchetChannel?
+        if channel.type == .singleRatchet {
+            ratchetChannel = try Virgil.ethree.getRatchetChannel(with: card)
+        } else {
+            ratchetChannel = nil
+        }
+
         let exported = try message.exportAsJsonData()
 
-        let card = try channel.getCard()
-        let ciphertext = try Virgil.ethree.authEncrypt(data: exported, for: card)
+        let ciphertext = try self.encrypt(data: exported, for: card, withRacthetChannel: ratchetChannel)
 
         var additionalData = additionalData
 
         if let data = additionalData {
-            additionalData = try Virgil.ethree.authEncrypt(data: data, for: card)
+            additionalData = try self.encrypt(data: data, for: card, withRacthetChannel: ratchetChannel)
         }
 
         let encryptedMessage = EncryptedMessage(ciphertext: ciphertext, date: date, additionalData: additionalData)
 
         try Ejabberd.shared.send(encryptedMessage, to: channel.name)
     }
 
+    private func sendService(message: Message, additionalData: Data?, to channel: Storage.Channel, date: Date) throws {
+        let card = try channel.getCard()
+
+        let exported = try message.exportAsJsonData()
+
+        let ciphertext = try self.encrypt(data: exported, for: card, withRacthetChannel: nil)
+
+        let encryptedMessage = EncryptedMessage(ciphertext: ciphertext, date: date, additionalData: nil)
+
+        try Ejabberd.shared.send(encryptedMessage, to: channel.name)
+    }
+
+    private func encrypt(data: Data, for card: Card, withRacthetChannel ratchetChannel: RatchetChannel?) throws -> Data {
```

Comment:
withRatchetChannel*
And is swift style it should be just `with:`, since we have specific parameter type